From 554d1c4cdfc261dc67c622ee7216a8fedaef4827 Mon Sep 17 00:00:00 2001
From: Rathees Kumar R Chinannan <quic_rrchinan@quicinc.com>
Date: Mon, 25 Sep 2023 14:19:59 +0530
Subject: [PATCH] mac80211: feed the link_id to cfg80211_ch_switch_notify

Currently, host hardcodes link_id value to 0 and sends to user space.
Instead of that use link id stored on ieee80211_link_data.

This change fixes the warning with invalid link ID on cfg80211_ch_switch_notify
function.

Signed-off-by: Rathees Kumar R Chinannan <quic_rrchinan@quicinc.com>
---
 net/mac80211/mlme.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/net/mac80211/mlme.c b/net/mac80211/mlme.c
index 480b516..5b5955b 100644
--- a/net/mac80211/mlme.c
+++ b/net/mac80211/mlme.c
@@ -1728,7 +1728,7 @@ static void ieee80211_chswitch_post_beacon(struct ieee80211_link_data *link)
 		return;
 	}
 
-	cfg80211_ch_switch_notify(sdata->dev, &link->reserved_chandef, 0);
+	cfg80211_ch_switch_notify(sdata->dev, &link->reserved_chandef, link->link_id);
 }
 
 void ieee80211_chswitch_done(struct ieee80211_vif *vif, bool success)
-- 
2.17.1

